packages/skeleton-react/src/lib/components/Segment/Segment.tsx

```diff
@@ -1,113 +1,102 @@
 'use client';
 
-import React, { createContext, useContext, useEffect, useState } from 'react';
+import React, { createContext, useContext, useId } from 'react';
+import * as radio from '@zag-js/radio-group';
+import { useMachine, normalizeProps } from '@zag-js/react';
 import type { SegmentContextState, SegmentProps, SegmentItemsProps } from './types.js';
+import { noop } from '$lib/internal/noop.js';
 
 // Contexts ---
 
 export const SegmentContext = createContext<SegmentContextState>({
-	value: '',
-	name: '',
-	onSelectionHandler: () => {}
+	api: {} as ReturnType<typeof radio.connect>
 });
 
 // Components ---
 
 const SegmentRoot: React.FC<SegmentProps> = ({
-	value = '',
-	name = '',
 	// Root
 	base = 'inline-flex items-stretch overflow-hidden',
 	background = 'preset-outlined-surface-200-800',
 	border = 'p-2',
-	flexDirection = 'flex-row', // vertical: flex-col
 	gap = 'gap-2',
 	padding = '',
 	rounded = 'rounded-container',
 	width = '',
 	classes = '',
+	// Indicator
+	indicatorBase = 'top-[var(--top)] left-[var(--left)] w-[var(--width)] h-[var(--height)]',
+	indicatorBg = 'preset-filled',
+	indicatorRounded = 'rounded',
+	indicatorClasses = '',
 	// Events
-	onChange,
+	onValueChange = noop,
 	// Children
-	children
+	children,
+	// Zag
+	...zagProps
 }) => {
-	function onSelectionHandler(newValue: string) {
-		value = newValue;
-		if (onChange) onChange(newValue);
-	}
+	const [state, send] = useMachine(
+		radio.machine({
+			id: useId(),
+			onValueChange: (details) => {
+				onValueChange(details.value);
+			}
+		}),
+		{ context: zagProps }
+	);
+	const api = radio.connect(state, send, normalizeProps);
 
 	// Set Context
-	const ctx = {
-		value,
-		name,
-		onSelectionHandler
-	};
+	const ctx = { api };
+
+	// Reactive
+	// FIXME: always returns vertical
+	// const rxOrientation = state.context.orientation === 'vertical' ? 'flex-col' : 'flex-row';
+	const rxOrientation = 'flex-row';
 
 	return (
 		<div
-			role="radiogroup"
-			className={`${base} ${flexDirection} ${background} ${border} ${padding} ${gap} ${rounded} ${width} ${classes}`}
+			{...api.getRootProps()}
+			className={`${base} ${rxOrientation} ${background} ${border} ${padding} ${gap} ${rounded} ${width} ${classes}`}
 			data-testid="segment"
 		>
+			{/* Indicator */}
+			<div {...api.getIndicatorProps()} className={`${indicatorBase} ${indicatorBg} ${indicatorRounded} ${indicatorClasses}`}></div>
+			{/* Items */}
 			<SegmentContext.Provider value={ctx}>{children}</SegmentContext.Provider>
 		</div>
 	);
 };
 
 const SegmentItem: React.FC<SegmentItemsProps> = ({
-	id,
 	value,
-	title,
-	disabled = false,
 	// Root
-	base = 'btn',
-	active = 'preset-filled',
-	hover = 'hover:preset-tonal',
+	base = 'btn cursor-pointer z-[1]',
 	classes = '',
 	// Label
 	labelBase = 'pointer-events-none',
+	labelActiveText = 'preset-filled',
 	labelClasses = '',
-	// Events
-	onClick,
 	// Children
 	children
 }) => {
```

Comment:
Let zag handle it by spreading rest props. This includes `value` and `disabled` props. Something you were wondering (disabling one specific item, instead of the whole segmented group).
```suggestion
const SegmentItem: React.FC<SegmentItemsProps> = ({
	// Root
	base = 'btn cursor-pointer z-[1]',
	classes = '',
	// Label
	labelBase = 'pointer-events-none',
	labelActiveText = 'preset-filled',
	labelClasses = '',
	// Children
	children,
    // Zag
    ...zagProps
}) => {
```

Reply:
...this is already done, right?

Reply:
You have `value` defined, and not `...zagProps`

Reply:
Oh on the `SegmentItem`. Yeah I wasn't sure if we wanted to handle that with the "overflow" zag components, as it's the only prop we're using in the items. What do you think?

Reply:
I don't think Zag supports `disabled` state for individual items. Only for the entire group, which is already handled in `<Segment>`:

- https://zagjs.com/components/react/segmented-control#disabling-the-segmented-control
- https://zagjs.com/components/react/segmented-control#making-the-segmented-control-readonly

This is set on the `machine`, which is only initialized per the overall "group". Unless I'm misunderstanding the docs.

Reply:
It does support it but it's undocumented, I tried it out when I was reviewing your code.
![image](https://github.com/user-attachments/assets/79b2479f-174a-4f5b-a818-255a30f6b6a7)

Reply:
`value` is required, `disabled` is what you wanted, `invalid` is something I'm not familiar with.

Reply:
How do I apply this within the `<SegmentItem>` though? There's no `machine` per item? The only data we're passing down from parent -> child component is the `api`. Can I provide these values to `api` somehow?

Reply:
Whenever you do `{ value }` just replace that with `zagProps`

Reply:
Same type errors described here: https://github.com/skeletonlabs/skeleton/pull/2778#discussion_r1707673468